module/extension/Director/tests/simple_task.cpp

```diff
@@ -0,0 +1,53 @@
+#include <catch.hpp>
+#include <nuclear>
+
+#include "Director.hpp"
+#include "TestBase.hpp"
+
+// Anonymous namespace to avoid name collisions
+namespace {
+
+    struct SimpleTask {};
+
+    std::vector<std::string> events;
+
+    class TestReactor : public TestBase<TestReactor> {
+    public:
+        explicit TestReactor(std::unique_ptr<NUClear::Environment> environment)
+            : TestBase<TestReactor>(std::move(environment)) {
+
+            on<Provide<SimpleTask>>().then([this] {
+                // Task has been executed!
+                events.push_back("task executed");
+            });
+
+            /**************
+             * TEST STEPS *
+             **************/
+            on<Trigger<Step<1>>, Priority::LOW>().then([this] {
+                events.push_back("emitting task");
+                emit<Task>(std::make_unique<SimpleTask>());
+            });
+            on<Startup>().then([this] {  //
```

Comment:
Looks like you were gonna put in a comment?

Reply:
I'm just tricking the formatter so it doesn't make a mess